client/src/App.js

```diff
@@ -7,42 +7,54 @@ import EMS from './EMS';
 import ER from './ER';
 import Admin from './Admin';
 import Redirect from './Components/Redirect';
+import Spinner from './Components/Spinner';
 
 function App() {
-  const { setUser, setOrganization, setHospital, setHospitalUser } = useContext(Context);
+  const { user, setUser, setOrganization, setHospital, setHospitalUser } = useContext(Context);
 
   useEffect(() => {
     // hit the users endpoint to ensure authenticated
     ApiService.users.me().then((response) => {
       // save the user data into the context
       const user = response.data;
-      setUser(user);
       setOrganization(user.organization);
-      if (user.organization.type === 'HEALTHCARE') {
+      if (user.organization?.type === 'HEALTHCARE') {
         // TODO: handle user added to multiple hospitals
-        setHospital(user.activeHospitals[0].hospital);
-        setHospitalUser(user.activeHospitals[0]);
+        if (user.activeHospitals?.length > 0) {
+          setHospital(user.activeHospitals[0].hospital);
+          setHospitalUser(user.activeHospitals[0]);
+        }
       }
+      setUser(user);
     });
   }, [setUser, setOrganization, setHospital, setHospitalUser]);
```

Comment:
I believe these state setters are guaranteed to never change, so it's not necessary to include them in the dependency array.

Reply:
I thought so too, but eslint complains when I remove them...?

Reply:
I guess eslint is too dumb to know that they won't change.  https://stackoverflow.com/a/71942752/4200446